Prepare for 4.0.0 RC2 (#1138)
fleetdm/fleet: CHANGELOG.md

* Add ability to separate hosts into exclusive groups with the Teams feature. The Teams feature is available for Fleet Basic customers. Check out the list below for the new functionality included with Teams:

* Teams: Add ability to enroll hosts to one team using team specific enroll secrets.

* Teams: Add ability to manually transfer hosts to a different team in the Fleet UI.

* Teams: Add ability to apply unique agent options to each team. Note that "osquery options" have been renamed to "agent options."

* Teams: Add ability to grant users access to one or more teams. This allows you to define a user's access to specific groups of hosts in Fleet.

* Add ability to create an API-only user. API-only users cannot access the Fleet UI. These users can access all Fleet API endpoints and `fleetctl` features. Available in Fleet Core.

* Add Redis cluster support. Available in Fleet Core.

* Fix a bug that prevented the columns chosen for the "Hosts" table from persisting after logging out of Fleet.

### Upgrade plan

Fleet 4.0.0 is a major release and introduces several breaking changes and database migrations. 

> * Use strictly `fleet` in Fleet's configuration, API routes, and environment variables. This means that you must update all usage of `kolide` in these items. The backwards compatibility introduced in Fleet 3.8.0 is no longer valid in Fleet 4.0.0.

Reviewer: ```suggestion
* Use strictly `fleet` in Fleet's configuration, API routes, and environment variables. Users must update all usage of `kolide` in these items (deprecated since Fleet 3.8.0).
```

> * Change configuration option `server_tlsprofile` to `server_tls_compatability`. This options previously had an inconsistent key name.

Reviewer: ```suggestion
* Change configuration option `server_tlsprofile` to `server_tls_compatability`. This option previously had an inconsistent key name.
```

* Replace the use of the `api/v1/fleet/spec/osquery/options` with `api/v1/fleet/config`. In Fleet 4.0.0, "osquery options" are now called "agent options." The new agent options are moved to the Fleet application config spec file and the `api/v1/fleet/config` API endpoint.

* Enroll secrets no longer have "names" and are now either global or for a specific team. Hosts no longer store the “name” of the enroll secret that was used. Users that want to be able to segment hosts (for configuration, queries, etc.) based on the enrollment secret should use the Teams feature in Fleet Basic.

* `auth_jwt_key` and `auth_jwt_key_file` are no longer accepted as configuration. 

* JWT encoding is no longer used for session keys. Sessions now default to expiring in 4 hours of inactivity.

### Known issues

> * ~~Query packs cannot be targeted to teams.~~
> * ~~Global agent options (osquery options cannot be updated in the Fleet UI.~~
> * ~~Migration order prevents upgrade from 3.13.0 to 4.0.0-rc1~~

Reviewer: ```suggestion
```
